operator/internal/manifests/storage/configure.go

```diff
@@ -126,139 +126,69 @@ func ensureObjectStoreCredentials(p *corev1.PodSpec, opts Options) corev1.PodSpe
 		MountPath: secretDirectory,
 	})
 
-	var storeEnvVars []corev1.EnvVar
-	switch storeType {
+	if !wifEnabled(opts) {
+		container.Env = append(container.Env, envVarsNoWIF(opts)...)
+	} else {
+		setSATokenPath(&opts, osEnabled)
+		container.Env = append(container.Env, envVarsWIF(opts)...)
+		volumes = append(volumes, saTokenVolume(opts))
+		container.VolumeMounts = append(container.VolumeMounts, saTokenVolumeMount(opts))
+	}
```

Comment:
I really like the style introduced here. It makes this function more readable. Although I feel `wif` as a prefix is a bit hard to grasp (as in general the whole managed identity federation concepts). I suggest the following rename that we should pass into the [enhancement proposal](https://github.com/grafana/loki/pull/11060):
1. `StaticAuthCredentials`: For everything that is using a static serviceaccount and accompanying credentials.
2. `ManagedAuthCredentials`: For everything that is using a federated/managed identity to access an authentication token at the end.

We can leave `Credentials` out if it yields too long function/variable names.